x-pack/legacy/plugins/maps/public/layers/vector_layer.js

```diff
@@ -206,11 +211,14 @@ export class VectorLayer extends AbstractLayer {
   }
 
   async getBounds(dataFilters) {
-    if (this._source.isBoundsAware()) {
-      const searchFilters = this._getSearchFilters(dataFilters);
-      return await this._source.getBoundsForFilters(searchFilters);
+
+    const isStaticLayer = !this._source.isBoundsAware() || !this._source.isFilterByMapBounds();
```

Comment:
this now will only ignore visibility of features for sources that are bounds aware (ie. we can actually query for the entire bounds), _and_ which are filtered based on extent.

E.g. Fit will take into account visibility for an es-doc source layer with all data in the view